### DISCUSSION TITLE
[err] Only anonymous FTP is allowed here

### DISCUSSION THREAD

**Discussion by Sannkulkarni** [stilliard/docker-pure-ftpd#177] [Q&A]
Hello all,

I'm trying to install and run the pure-ftpd docker image in kubernetes as pods using a customized Dockerfile as per the guidelines mentioned here ( https://hub.docker.com/r/stilliard/pure-ftpd/). However, when i try to connect to the pure-ftpd server with some ftp client, I'm able to connect successfully but not able to login and browse the files etc. 

Here is the list of messages seen on the ftp client console :- 

Response:	220---------- Welcome to Pure-FTPd [privsep] [TLS] ----------
Response:	220-You are user number 4 of 10 allowed.
Response:	220-Local time is now 11:31. Server port: 21.
Response:	220-Only anonymous FTP is allowed here
Response:	220-IPv6 connections are also welcome on this server.
Response:	220 You will be disconnected after 15 minutes of inactivity.
Command:	AUTH TLS
Response:	500 This security scheme is not implemented
Command:	AUTH SSL
Response:	500 This security scheme is not implemented
Status:	Insecure server, it does not support FTP over TLS.
Command:	USER xxxxx
Response:	421 Unable to set up secure anonymous FTP
Error:	Could not connect to server

**Dockerfile** (please note that I'm using -E and required UN and Password for 'NoAnonymous' option)

<snip>
FROM stilliard/pure-ftpd

CMD /run.sh -c 10 -C 15 -l puredb:/etc/pure-ftpd/pureftpd.pdb -E -j -R -P $PUBLICHOST -p 30000:30019 -e FTP_USER_NAME=xxxxx -e FTP_USER_PASS=xxxxx -e FTP_USER_HOME=/home/xxxxx -e "ADDED_FLAGS=-d -d"
<snip>

**docker-compose** (please note that I'm using required UN and Password here too :) )

<snip>
ftpd_server_2:
image: my-pure-ftpd:v12
container_name: "pure-ftpd"
ports:
- "21:21"
- "30000-30019:30000-30019"
 .
 .
environment:
PUBLICHOST: xxxxx
FTP_USER_NAME: xxxxx
FTP_USER_PASS: xxxxx
FTP_USER_HOME: /home/xxxxx

restart always
<snip>

Not sure if I'm missing anything here, any leads to above issue would be much appreciated.

TIA :)

**Comment by stilliard** [comment]
Hi @Sannkulkarni, sorry for the delyed reply. 

I think the env vars (-e FTP_USER_NAME=xxxxx etc.) need to passed when calling docker run or in your docker compose instead of inside the Dockerfile.

Atm you're passing these as flags to pure-ftpd itself which `-e` is `-e --anonymousonly` ref https://linux.die.net/man/8/pure-ftpd

Hope this helps.
